packages/@lwc/wds-core/src/node/plugins/lwc/compile.js

```diff
@@ -89,15 +89,19 @@ export default ({ rootDir, moduleDirs }) => ({
   },
 
   resolveImport({ source, context, code, column, line }) {
-    const filePath = getRequestFilePath(context.url, rootDir);
+    const parentModuleFilePath = getRequestFilePath(context.url, rootDir);
+
     const isRelativeImport = source.startsWith('.');
-    const resolvedImport = path.resolve(path.dirname(filePath), source);
+    const resolvedImport = path.resolve(
+      path.dirname(parentModuleFilePath),
+      stripQueryString(source),
```

Comment:
The `.css` or `.html` detection below doesn't work unless you remove the query string from the resolve import.